src/script/miniscript.h

```diff
@@ -301,8 +303,13 @@ struct Node {
     const Type typ;
     //! Cached script length (computed by CalcScriptLen).
     const size_t scriptlen;
-    //! Whether a public key appears more than once in this node.
-    const bool duplicate_key;
+    //! Whether a public key appears more than once in this node. This value is initialized
+    //! by all constructors except the NoDupCheck ones. The NoDupCheck ones skip the
+    //! computation, requiring it to be done manually by invoking DuplicateKeyCheck().
+    //! DuplicateKeyCheck(), or a non-NoDupCheck constructor, will compute has_duplicate_keys
+    //! for all subnodes as well.
```

Comment:
I'm assuming this is useful because it allows performing other checks (like size limit) _before_ checking for duplicates? Is so, maybe say that in the comment.

Reply:
It could technically be used for this but it's not the intended purpose: we only ever use it to skip redundant computation.